app/routers/wallet.py

```diff
@@ -16,15 +18,106 @@
 
 admin_url = os.getenv("ACAPY_ADMIN_URL")
 admin_port = os.getenv("ACAPY_ADMIN_PORT")
+# TODO Should the admin_api_key be a dummy variable so the controller doesn't function w/o providing it?
+# This all smells really - this has to be done in a better manner
 admin_api_key = os.getenv("ACAPY_ADMIN_API_KEY")
 is_multitenant = os.getenv("IS_MULTITENANT", False)
 ledger_url = os.getenv("LEDGER_NETWORK_URL")
 
 
-@router.get(
-    "/create-pub-did", tags=["did"], response_model=DidCreationResponse
-)
-async def create_public_did():
+async def __create_did(controller):
+    generate_did_res = await controller.wallet.create_did()
+    if not generate_did_res["result"]:
+        controller.terminate()
```

Comment:
It is not an issue for this actual code - actually for the cloudcontroller context... We should see if we can use the context manager pattern in python to manage the controller... 

Then thix contesxt would go
```
with controller.wallet.create_did() as generate_did_res: 
    // do the work with created did
```

That way the controller wallet can manage it's own termination etc

If we did that this code would not need to know it needs to terminate. IT would just use the with context and that would be handled

Reply:
yeah , context manager could be good here, but If I'm not missing the obvious that's require changing the cloudcontroller to have `__enter__` and `__exit__` methods?

Also:
`with controller.wallet.create_did() as generate_did_res:` needs ot at least be async with

I guess we could also create a helper class that just instatiates the controller in a contextmanager capapble fashion?

Reply:
yes - would only do it if it can be done inside the cloudcontroller library. Would strongly prefer not to wrap it in this context. Probably best if we introduce another method that creates it as a context manager

Reply:
hmmm made a mistake with my code example...

would be something more like 

```
with get_wallet_controller() as wallet_controller:
 generate_did_res = await controller.create_did()
 // carry on with processing

```